app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.kt

```diff
@@ -0,0 +1,272 @@
+     */
+    private val allowedCardAuthMethods = JSONArray(Constants.SUPPORTED_METHODS)
+
+    /**
+     * Describe your app's support for the CARD payment method.
+     *
+     *
+     * The provided properties are applicable to both an IsReadyToPayRequest and a
+     * PaymentDataRequest.
+     *
+     * @return A CARD PaymentMethod object describing accepted cards.
+     * @throws JSONException
+     * @see [PaymentMethod](https://developers.google.com/pay/api/android/reference/object.PaymentMethod)
+     */
+    // Optionally, you can add billing address/phone number associated with a CARD payment method.
+    private fun baseCardPaymentMethod(): JSONObject {
+        val cardPaymentMethod = JSONObject()
+        cardPaymentMethod.put("type", "CARD")
+
+        val parameters = JSONObject()
+        parameters.put("allowedAuthMethods", allowedCardAuthMethods)
+        parameters.put("allowedCardNetworks", allowedCardNetworks)
+        parameters.put("billingAddressRequired", true)
+
+        val billingAddressParameters = JSONObject()
+        billingAddressParameters.put("format", "FULL")
+
+        parameters.put("billingAddressParameters", billingAddressParameters)
+
+        cardPaymentMethod.put("parameters", parameters)
+
+        return cardPaymentMethod
+    }
+
+    /**
+     * Describe the expected returned payment data for the CARD payment method
+     *
+     * @return A CARD PaymentMethod describing accepted cards and optional fields.
+     * @throws JSONException
+     * @see [PaymentMethod](https://developers.google.com/pay/api/android/reference/object.PaymentMethod)
+     */
+    private fun cardPaymentMethod(): JSONObject {
+        val cardPaymentMethod = baseCardPaymentMethod()
+        cardPaymentMethod.put("tokenizationSpecification", gatewayTokenizationSpecification())
+
+        return cardPaymentMethod
+    }
+
+    /**
+     * An object describing accepted forms of payment by your app, used to determine a viewer's
+     * readiness to pay.
+     *
+     * @return API version and payment methods supported by the app.
+     * @see [IsReadyToPayRequest](https://developers.google.com/pay/api/android/reference/object.IsReadyToPayRequest)
+     */
+    fun isReadyToPayRequest(): Optional<JSONObject> {
+        try {
+            val isReadyToPayRequest = baseRequest
+            isReadyToPayRequest.put(
+                    "allowedPaymentMethods", JSONArray().put(baseCardPaymentMethod()))
+
+            return Optional.of(isReadyToPayRequest)
+        } catch (e: JSONException) {
+            return Optional.empty()
+        }
+    }
+
+    /**
+     * Information about the merchant requesting payment information
+     *
+     * @return Information about the merchant.
+     * @throws JSONException
+     * @see [MerchantInfo](https://developers.google.com/pay/api/android/reference/object.MerchantInfo)
+     */
+    private val merchantInfo: JSONObject
+        @Throws(JSONException::class)
```

Comment:
If you remove dependency on  org.json. then you save all of us from distracting JSONException everywhere